controllers/storagecluster/storagecluster_controller.go

```diff
@@ -128,16 +128,6 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	enqueueStorageClusterRequest := handler.EnqueueRequestsFromMapFunc(
 		func(context context.Context, obj client.Object) []reconcile.Request {
-
-			ocinit, ok := obj.(*ocsv1.OCSInitialization)
-			if !ok {
-				return []reconcile.Request{}
-			}
-
-			if ocinit.Status.Phase != util.PhaseReady {
-				return []reconcile.Request{}
-			}
-
```

Comment:
The logic here is trying to be both an enqueue func and a filter func, which means the original author was using the wrong tool for the job.
The intent was, as I read it, NOT to enqueue an event for a storage cluster if the OCS init resource is not at the ready phase. We cannot just remove that logic as there must be a reason the logic was introduced in the first place (maybe it solves a different bug) 

I agree that the checks need to be removed here as this enqueue func is being used in watches other than ocs init watches. But we do need to keep the logic intact for ocs init resource.

A more correct fix will be to introduce a new predicate func that assumes OCSInitialization CR and filter the events if the ocs init CR is not in ready phase. Then add that predicate to the ocs init watch. This will be done on top of the removal of these lines from the above function.

Reply:
Acc to the discussion we had with @iamniting who had initially added the check for ocsinit resource being ready, this check no longer makes sense now, @iamniting can shed more light on the reasoning. Full discussion here https://redhat-internal.slack.com/archives/CJW8YBDR7/p1726739675199359?thread_ts=1726559338.202149&cid=CJW8YBDR7.

Just FYI this code has already been removed in later releases 4.15+.

Reply:
ack